### PR TITLE
Adding Smartseq2, Dockstore yaml, and Updating to Relative imports

### PR DESCRIPTION
**Pull request by kbergin**
To support easy release from Dockstore to Terra, we need to use relative imports relative to where the WDL is. I've tried to do this here. 

I've also read that [Dockstore can have automatic updates](https://docs.dockstore.org/en/develop/getting-started/github-apps/github-apps.html) to workflows from github if you add workflows using a github app and add a .dockstore.yaml. It seems as though it can support multiple separate workflows from one github repo. 

To test out the multiple pipelines support for releasing, I have also added smartseq2 single and multi sample pipelines (with updated relative imports). I pulled these from skylab master on 07/20 - edited to add, I re-pulled from skylab master on 7/28, which seemed up to date with the latest smartseq2 releases.

### CODE REVIEW

File: pipelines/skylab/smartseq2_multisample/MultiSampleSmartSeq2.wdl [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+      # Sample information
+      String stranded
+      String file_prefix
+      Array[String] input_file_names
+      String batch_id
+      Boolean paired_end
+  }
+  # Version of this pipeline
+  String pipeline_version = "MultiSampleSmartSeq2_v2.0.1"
```
Comment (kbergin) [inline pipelines/skylab/smartseq2_multisample/MultiSampleSmartSeq2.wdl:32]: Change this to mimic dsde-pipelines where the name of pipeline isn't included in the version variable

File: pipelines/skylab/smartseq2_multisample/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |
+| :----: | :---: | :----: | :--------------: |
+| v.2.0.0 | June, 2020 | [Elizabeth Kiernan](mailto:ekiernan@broadinstitute.org) | Please file GitHub issues in skylab or contact [Kylee Degatano](mailto:kdegatano@broadinstitute.org) |
```
Comment (kbergin) [inline pipelines/skylab/smartseq2_multisample/README.md:3]: Read through readme to find areas that need to be updated for new repo

File: pipelines/skylab/smartseq2_single_sample/Creating_Smartseq2_References.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# This Document is Under Construction
+
```
Comment (kbergin) [inline pipelines/skylab/smartseq2_single_sample/Creating_Smartseq2_References.md:2]: double check if there is a latest of this document or talk to Liz

File: pipelines/skylab/smartseq2_single_sample/README.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
+| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |
+| :----: | :---: | :----: | :--------------: |
+| [smartseq2_v4.0.0](https://github.com/HumanCellAtlas/skylab/releases/tag/smartseq2_v3.0.0) | June, 2020 | [Elizabeth Kiernan](mailto:ekiernan@broadinstitute.org) | Please file GitHub issues in skylab or contact [Kylee Degatano](mailto:kdegatano@broadinstitute.org) |
```
Comment (kbergin) [inline pipelines/skylab/smartseq2_single_sample/README.md:3]: Same comment as other readme, read through to update

File: pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    String stranded
+    String sample_name
+    String output_name
+    File fastq1
+    File? fastq2
+    Boolean paired_end
+    Boolean force_no_check = false
+  }
+  # version of this pipeline
+  String pipeline_version = "smartseq2_v4.0.1"
```
Comment (kbergin) [inline pipelines/skylab/smartseq2_single_sample/SmartSeq2SingleSample.wdl:37]: Change to just version number not full name

File: pipelines/skylab/smartseq2_single_sample/options.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "monitoring_script": "gs://broad-dsde-mint-dev-teststorage/10x/benchmark/scripts/monitor.sh",
```
Comment (kbergin) [inline pipelines/skylab/smartseq2_single_sample/options.json:2]: Change to same monitoring script to the one used in dsde-pipelines

**Comment by kbergin** [comment]
Add docker images

File: scripts/build_workflow_release.sh [hunk 7]
```diff
@@ -42,14 +41,17 @@ function deploy_options() {
   local base_options
   if [ -f ${wdl_dir}/${prefix}.options.json ]; then
     base_options=${prefix}.options.json
+    cp ${wdl_dir}/${base_options} ${target_dir}/${versioned_options}
   elif [ -f ${wdl_dir}/${prefix}.${env}.options.json ]; then
     base_options=${prefix}.${env}.options.json
+    cp ${wdl_dir}/${base_options} ${target_dir}/${versioned_options}
+  elif [ -f ${DSDE_PIPELINES_ROOT}/tests/skylab/test.options.json ]; then
+    cp ${DSDE_PIPELINES_ROOT}/tests/skylab/test.options.json ${target_dir}/${versioned_options}
```
Comment (jessicaway) [inline scripts/build_workflow_release.sh:49]: Maybe we should change this to check if the wdl directory path includes `skylab` so we don't use these options for other pipelines? 
Also we should be able to just set the `base_options` and then do the copy once. 
```suggestion
    base_options=${DSDE_PIPELINES_ROOT}/tests/skylab/test.options.json
```

File: scripts/build_workflow_release.sh [hunk 8]
```diff
@@ -42,14 +41,17 @@ function deploy_options() {
...
   else
     echo >&2 Error: Options JSON not found at either ${prefix}.options.json or ${prefix}.${env}.options.json
     exit 1
   fi
 
-  cp ${wdl_dir}/${base_options} ${target_dir}/${versioned_options}
```
Comment (jessicaway) [inline scripts/build_workflow_release.sh:52]: If you set base_options for all tree paths, you can keep this here and remove the other instances above.

File: tests/skylab/trigger_test.sh [hunk 9]
```diff
@@ -9,10 +9,30 @@ TEST_TYPE=${2:-pr}
 TEST_TYPE_UC=$(printf $TEST_TYPE | awk '{ print toupper($0) }')
 WD=$(pwd)
 
+TEST_WDL_FILE="${WD}/tests/skylab/${PIPELINE_FOLDER_NAME}/${TEST_TYPE}/test_${PIPELINE_FOLDER_NAME}_${TEST_TYPE_UC}.wdl"
 
-INPUTS_JSON="/working/tests/skylab/${PIPELINE_FOLDER_NAME}/${TEST_TYPE}/test_inputs.json"
-WDL_FILE="/working/tests/skylab/${PIPELINE_FOLDER_NAME}/${TEST_TYPE}/test_${PIPELINE_FOLDER_NAME}_${TEST_TYPE_UC}.wdl"
-DEPENDENCIES_JSON="/working/tests/skylab/${PIPELINE_FOLDER_NAME}/${TEST_TYPE}/dependencies.json"
+echo "calling release script to flatten dependencies and imports"
```
Comment (jessicaway) [inline tests/skylab/trigger_test.sh:14]: Might not need the echo here

Comment (kbergin) [reply tests/skylab/trigger_test.sh]: Changed it to a comment, thanks!

File: scripts/build_workflow_release.sh (hunk 8 above)
Comment (kbergin) [reply scripts/build_workflow_release.sh:52]: I think I made this change work, if you want to take a second look.

File: scripts/build_workflow_release.sh [hunk 10]
```diff
@@ -41,15 +40,17 @@ function deploy_options() {
   # Some workflows have per-environment options, and others share options across all environments.
   local base_options
   if [ -f ${wdl_dir}/${prefix}.options.json ]; then
-    base_options=${prefix}.options.json
+    base_options=${wdl_dir}/${prefix}.options.json
   elif [ -f ${wdl_dir}/${prefix}.${env}.options.json ]; then
-    base_options=${prefix}.${env}.options.json
+    base_options=${wdl_dir}/${prefix}.${env}.options.json
+  elif [[ "${wdl_dir}" =~ .*"skylab".* ]]; then
+    base_options=${DSDE_PIPELINES_ROOT}/tests/skylab/test.options.json
```
Comment (jessicaway) [inline scripts/build_workflow_release.sh:47]:
```suggestion
    base_options=${DSDE_PIPELINES_ROOT}/tests/skylab/options.json
```

Comment (kbergin) [reply scripts/build_workflow_release.sh:47]: I thought I changed the name of the file in that folder to test.options.json though?

File: scripts/build_workflow_release.sh (hunk 8 above)
Comment (jessicaway) [reply scripts/build_workflow_release.sh:52]: looks good!

File: scripts/build_workflow_release.sh (hunk 10 above)
Comment (jessicaway) [reply scripts/build_workflow_release.sh:47]: you did, you're right! Ignore me

File: scripts/build_workflow_release.sh (hunk 7 above)
Comment (kbergin) [reply scripts/build_workflow_release.sh]: Done!

**Pull request merged**